simulator/src/main/java/com/hazelcast/simulator/common/SimulatorProperties.java

```diff
@@ -16,82 +16,61 @@
- * as defaults. If a simulator.properties is available in the working dir of if an explicit simulator.properties
- * is configured, it will override the properties from the default.
+ * Loads the Hazelcast Simulator properties file.
+ *
+ * This class will always load the properties in the <tt>${SIMULATOR_HOME}/conf/simulator.properties</tt> as defaults. If an
+ * explicit <tt>simulator.properties</tt> file is configured or <tt>simulator.properties</tt> is available in the working dir,
+ * it will override the properties from the default.
  */
 public class SimulatorProperties {
+
+    static final String PROPERTIES_FILE_NAME = "simulator.properties";
+
     private static final Logger LOGGER = Logger.getLogger(SimulatorProperties.class);
 
     private final Properties properties = new Properties();
+
     private String forcedHazelcastVersionSpec;
 
     public SimulatorProperties() {
-        File defaultPropsFile = newFile(getSimulatorHome(), "conf", "simulator.properties");
-        LOGGER.debug("Loading default simulator.properties from: " + defaultPropsFile.getAbsolutePath());
+        File defaultPropsFile = newFile(getSimulatorHome(), "conf", PROPERTIES_FILE_NAME);
+        LOGGER.debug(format("Loading default %s from: %s", PROPERTIES_FILE_NAME, defaultPropsFile.getAbsolutePath()));
+        check(defaultPropsFile);
         load(defaultPropsFile);
     }
 
-    public String getUser() {
-        return get("USER", "simulator");
-    }
-
-    public boolean isEc2() {
-        return "aws-ec2".equals(get("CLOUD_PROVIDER"));
-    }
-
-    public String getHazelcastVersionSpec() {
-        if (forcedHazelcastVersionSpec == null) {
-            return get("HAZELCAST_VERSION_SPEC", "outofthebox");
-        } else {
-            return forcedHazelcastVersionSpec;
-        }
-    }
-
-    public void forceGit(String gitRevision) {
-        if (gitRevision != null && !gitRevision.isEmpty()) {
-            forcedHazelcastVersionSpec = HazelcastJars.GIT_VERSION_PREFIX + gitRevision;
-            LOGGER.info("Overriding Hazelcast version to GIT revision " + gitRevision);
-        }
-    }
-
     /**
-     * Initialized the SimulatorProperties
+     * Initializes the SimulatorProperties.
      *
      * @param file the file to load the properties from. If the file is null, then first the simulator.properties
      *             in the working dir is checked.
      */
     public void init(File file) {
         if (file == null) {
-            //if no file is explicitly given, we look in the working directory
-            File fallbackPropsFile = new File("simulator.properties");
-            if (fallbackPropsFile.exists()) {
-                file = fallbackPropsFile;
-            } else {
-                LOGGER.warn(format("%s is not found, relying on defaults", fallbackPropsFile));
+            // if no file is explicitly given, we look in the working directory
```

Comment:
I found the old method very hard to read. I hope I didn't change the logic with this rewrite, should still be the same.